lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -382,24 +393,34 @@ export default function getRecipeInterface(
             logDebugMessage("refreshSession: Started");
             // We use a fallback mechanism here, to ensure there is a smooth upgrade path when switching transfer methods
```

Comment:
to make code review easier for this function, you should add a comment explaining all the possible inputs and the expected output like:
```
/*
Case 1:
input:
   - auth-method header: cookie
   - cookie: refreshtoken
   - authorisationheader: undefined
output:
   - new tokens in cookies

Case 2:
input:
   - auth-method header: header
   - cookie: refreshtoken
   - authorisationheader: undefined
output:
   - new tokens in headers 

Case 3:
input:
   - auth-method header: undefined
   - cookie: refreshtoken
   - authorisationheader: undefined
output:
   - new tokens in cookies

and so on.. (including what happens if both cookies and auth headers are present)
*/
```